cargo-auditable/src/collect_audit_data.rs

```diff
@@ -4,13 +4,33 @@ use std::str::from_utf8;
 
 use crate::{
     auditable_from_metadata::encode_audit_data, cargo_arguments::CargoArgs,
-    rustc_arguments::RustcArgs,
+    rustc_arguments::RustcArgs, sbom_precursor,
 };
 
 /// Calls `cargo metadata` to obtain the dependency tree, serializes it to JSON and compresses it
 pub fn compressed_dependency_list(rustc_args: &RustcArgs, target_triple: &str) -> Vec<u8> {
-    let metadata = get_metadata(rustc_args, target_triple);
-    let version_info = encode_audit_data(&metadata).unwrap();
+    let sbom_path = std::env::var_os("CARGO_SBOM_PATH");
+
+    // If cargo has created precursor SBOM files, use them instead of `cargo metadata`.
+    let version_info = if sbom_path.as_ref().map(|p| !p.is_empty()).unwrap_or(false) {
+        let sbom_paths = std::env::split_paths(&sbom_path.unwrap()).collect::<Vec<_>>();
+        // Cargo may create multiple SBOM precursor files.
+        // We can't control per-binary (or cdylib) dependency information, just grab the first non-rlib SBOM we find.
```

Comment:
If I recall correctly, this is only reached when running a `rustc` command to compile a specific crate. Since dependencies are crate-wide, but a crate can have multiple targets (rlib, bin, cdylib, etc) it shouldn't really matter which SBOM we find - they should all contain identical dependency trees.

If I'm mistaken, I'd really like to know about that. And if my understanding is correct, we should include this as a comment.

Reply:
Does cargo emit an SBOM file per-target or per-binary? A single lib target can result in multiple binaries for each different specified library crate type through a single rustc invocation.